src/metabase/automagic_dashboards/adaptive_dashboard_size.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.automagic-dashboards.adaptive-dashboard-size
+  (:require [bigml.histogram.core :as hist]
+            [clojure.core.memoize :as memo]
+            [metabase.models
+             [database :refer [Database]]
+             [query-execution :refer [QueryExecution]]]
+            [metabase.sync.analyze.fingerprint.fingerprinters :refer [histogram]]
+            [redux.core :as redux]
+            [toucan.db :as db]))
+
+(def ^:private ^Long long-running-90th-percentile-threshold
```

Comment:
As mentioned off-line, I don't like the idea of x-rays having their own sense of whether a given card/database is slow. 

This should be centralized somewhere other than the automagic dashboard code.